users/jdelkins/jdelkins.h

```diff
@@ -0,0 +1,142 @@
+/*
+  Copyright 2020 Joel Elkins <joel@elkins.com>
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "quantum.h"
+
+// Secrets
+
+#ifndef NO_SECRETS
+#  if __has_include("secrets.h")
+#    include "secrets.h"
+#    define DO_SECRETS
+     void send_secret_string(uint8_t n);
+#  else
+#    pragma message("Warning: secrets.h not found")
+#    define send_secret_string(_)
+#  endif
+#else
+#  define send_secret_string(_)
+#endif
+
+// standard layers
+
+enum jdelkins_layers {
+    _QWERTY = 0,
+    _RPT,
+    _GAME,
+    _FUNC,
+    _KP,
+    _SECRETS,
+    _ADJUST,
+    _LAYER_MAX
+};
+
+// key definitions
+
+typedef union {
+        uint32_t raw;
+        struct {
+                bool system_mac :1;
+        };
+} user_config_t;
+
+extern user_config_t user_config;
+
+static inline void do_mac_key(uint16_t norm_key, uint16_t mac_key, keyrecord_t *record) {
+    uint16_t key = user_config.system_mac ? mac_key : norm_key;
```

Comment:
you could also use `keymap_config.lctrl_lgui_swap`, as this is the CG_SWAP stuff, which is very useful on mac. 

This is just a suggestion, though. I personally find it easier to use this, as it globally available.

Reply:
If it's the same to you, I think I'd prefer to keep it this way. I don't use bootmagic and using `keymap_config` instead of `user_config` without bootmagic doesn't get me anything.

Reply:
That's absolutely fine!  This was much more of a suggestion, based on usage.